[14.5-stable] Cleanup replica sets

zedi-pramodh opened the pull request:
This commit fixes a serious bug in cluster configuration. Basically whenever domainmgr restarts there was a legacy code which deletes all VMs running. That was implemented to handle a corner case where apps were deleted when device is powered off. We want to delete any app config in kubernetes as soon as device reboots. Assumption is device will resync config with the controller.

That works fine for single node cases. In a cluster setup, apps would have moved to other nodes when this device went to reboot. Because of this bug as soon as this device reboots its deleting VMs running on other nodes, basically bouncing all VMs.

Added a new API to check if this device is in cluster mode and ignore delete of VMs.

Also, we no longer directly deal with VMs, its all VM replica sets now. So moved CleanupVMs to CleanupVMRs.

Move the const to types/locationconstants.go
Change the logging to Debugf

Signed-off-by: Pramodh Pallapothu <pramodh@zededa.com>
(cherry picked from commit 43d4287b194dc01c2bb4f8d4fba7da0f071af070)

Backport of PR #5060

OhmSpectator commented:
Kubevirt build is failing

OhmSpectator reviewed (changes requested):
Kubevirt build is broken

zedi-pramodh commented:
That's strange since it is just a cherry pick. Let me check

OhmSpectator commented:
> That's strange since it is just a cherry pick. Let me check

Sometimes we merge a PR that breaks something, and later we fix it with another PR. @andrewd-zededa, @rene, could it be the case here?

zedi-pramodh commented:
ok it looks like the master branch has a new API in the file I changed, which is not in 14.5-stable branch. I removed that and amended the PR. Should be fine now.

zedi-pramodh commented on pkg/pillar/types/locationconsts.go:
```diff
@@ -141,6 +141,10 @@ const (
 	// CustomOVMFSettingsDir - directory for custom OVMF settings (for different resolutions)
 	CustomOVMFSettingsDir = "/hostfs/etc/ovmf"
+	// LocalActiveAppConfigDir - directory to put JSON of the apps that are running.
+	LocalActiveAppConfigDir = "/persist/vault/active-app-instance-config/"
```
That's correct. Its only const, can't we keep it in 14.5-stable branch ?

OhmSpectator replied:
Let @milan-zededa decide if we can leave the line in this PR =)

milan-zededa replied:
@zedi-pramodh Even if harmless, this code is from different commit so remove it from your cherry-pick.

eriknordmark replied:
+1

eriknordmark commented:
Both the yetus and spdx check failures are saying the PR needs to be rebased. Can you please do that and remove the unrelated change?

zedi-pramodh commented:
Ok done, removed the extra constant and rebased to 14.5-stable branch
